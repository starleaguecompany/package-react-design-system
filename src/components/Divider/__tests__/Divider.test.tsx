import * as React from 'react';
import { render, screen } from '@testing-library/react';

import { Divider } from '../../../index';

describe('Divider', () => {
  test('should render correctly with no props', () => {
    render(<Divider />);

    expect(screen.getByTestId('Divider')).toMatchSnapshot();
    expect(screen.getByTestId('Divider')).toHaveClass('container', { exact: true });
  });

  test('should render correctly with attributes', () => {
    render(<Divider id="test-id" className="test-class" />);

    expect(screen.getByTestId('Divider')).toMatchSnapshot();
    expect(screen.getByTestId('Divider')).toHaveClass('test-class');
    expect(screen.getByTestId('Divider')).toHaveAttribute('id', 'test-id');
  });

  test('should render correctly with size prop', () => {
    render(<Divider size={2} />);

    expect(screen.getByTestId('Divider')).toMatchSnapshot();
    expect(screen.getByTestId('Divider')).toHaveClass('size-2');
  });
});
