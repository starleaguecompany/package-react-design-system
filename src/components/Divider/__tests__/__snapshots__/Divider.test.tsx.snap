// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider should render correctly with attributes 1`] = `
<div
  class="test-class container"
  data-qa="Divider"
  id="test-id"
/>
`;

exports[`Divider should render correctly with no props 1`] = `
<div
  class="container"
  data-qa="Divider"
/>
`;

exports[`Divider should render correctly with size prop 1`] = `
<div
  class="container size-2"
  data-qa="Divider"
/>
`;
